### PR TITLE
Barebone hermes integration

### PR DESCRIPTION
**Pull request by mganandraj**
This change introduce Hermes JavaScript engine to react-native-windows repo. There is some more work required for seamless using Hermes by the apps though,
1. Hermes don't work correctly with haul created dev bundles.
2. Bytecode loading pipeline needs to be figured out.
3. Hermes doesn't load the JavaScript bundles of the Universal sample app because of the presence of webviews in it.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2896)

### CODE REVIEW

**Comment by tudorms** [comment]
Do we want to enable it for ReactWin32 as well?

File: vnext/ReactUWP/ReactUWP.vcxproj [hunk 1]
```diff
@@ -403,13 +407,18 @@
   <ImportGroup Label="ExtensionTargets">
     <Import Project="$(SolutionDir)packages\boost.1.68.0.0\build\boost.targets" Condition="Exists('$(SolutionDir)packages\boost.1.68.0.0\build\boost.targets')" />
     <Import Project="$(SolutionDir)packages\boost_date_time-vc141.1.68.0.0\build\boost_date_time-vc141.targets" Condition="Exists('$(SolutionDir)packages\boost_date_time-vc141.1.68.0.0\build\boost_date_time-vc141.targets')" />
+    <Import Project="..\packages\vcpkg-export-20190805-175352.1.0.0\build\native\vcpkg-export-20190805-175352.targets" Condition="Exists('..\packages\vcpkg-export-20190805-175352.1.0.0\build\native\vcpkg-export-20190805-175352.targets')" />
```
Comment (tudorms) [inline vnext/ReactUWP/ReactUWP.vcxproj]: >vcpkg-export-20190805-175352.1.0.0 [](start = 33, length = 34)

This appears to be missing from packages.config; is it still needed?

**Comment by acoates-ms** [comment]
I assume the haul thing is just a config issue with haul, nothing wrong with this specific PR.
Why would hermes affect bundles with webviews in it?  -- Note webviews currently through up a yellowbox, so something needs to be done there anyway.

**Comment by mganandraj** [comment]
> I assume the haul thing is just a config issue with haul, nothing wrong with this specific PR.
> Why would hermes affect bundles with webviews in it?  -- Note webviews currently through up a yellowbox, so something needs to be done there anyway.
Yep. We should configure Haul to not use evals in the dev bundles. The weird thing is that Hermes currently don't fail when JS contains evals. It has implemented evals in a conconformat ways, i.e. without capturing the variable from eval executing environment. Currently, when I try to load the sample app dev bundle, Hermes keeps loading the bundle forever, very quickly growing the memory footprint to more than 1G and them throwing OOM (it is a security risk)!! Hermes has a build time flag to totally disable "eval" which we should set when we take the next Hermes drop and build ..

On Webviews, it was my wild guess. Last when I tested, when loading Hermes bytecode, the sample app failed with a callstack in "chakra.dll" and it looked like happening from a webview control in it. Hence, I guessed that the bytecode conversion screwed up the webview inputs somehow. That guess could be totally wrong. I'll investigate further tomorrow.

**Comment by mganandraj** [comment]
> I assume the haul thing is just a config issue with haul, nothing wrong with this specific PR.
> Why would hermes affect bundles with webviews in it?  -- Note webviews currently through up a yellowbox, so something needs to be done there anyway.
Yep. We should configure Haul to not use evals in the dev bundles. The weird thing is that Hermes currently don't fail when JS contains evals. It has implemented evals in a conconformat ways, i.e. without capturing the variable from eval executing environment. Currently, when I try to load the sample app dev bundle, Hermes keeps loading the bundle forever, very quickly growing the memory footprint to more than 1G and them throwing OOM (it is a security risk)!! Hermes has a build time flag to totally disable "eval" which we should set when we take the next Hermes drop and build ..
> 
> On Webviews, it was my wild guess. Last when I tested, when loading Hermes bytecode, the sample app failed with a callstack in "chakra.dll" and it looked like happening from a webview control in it. Hence, I guessed that the bytecode conversion screwed up the webview inputs somehow. That guess could be totally wrong. I'll investigate further tomorrow.

**Comment by mganandraj** [comment]
> I assume the haul thing is just a config issue with haul, nothing wrong with this specific PR.
> Why would hermes affect bundles with webviews in it?  -- Note webviews currently through up a yellowbox, so something needs to be done there anyway.

Yep. We should configure Haul to not use evals in the dev bundles. The weird thing is that Hermes currently don't fail when JS contains evals. It has implemented evals in a conconformat ways, i.e. without capturing the variable from eval executing environment. Currently, when I try to load the sample app dev bundle, Hermes keeps loading the bundle forever, very quickly growing the memory footprint to more than 1G and them throwing OOM (it is a security risk)!! Hermes has a build time flag to totally disable "eval" which we should set when we take the next Hermes drop and build ..

On Webviews, it was my wild guess. Last when I tested, when loading Hermes bytecode, the sample app failed with a callstack in "chakra.dll" and it looked like happening from a webview control in it. Hence, I guessed that the bytecode conversion screwed up the webview inputs somehow. That guess could be totally wrong. I'll investigate further tomorrow.

File: vnext/ReactUWP/ReactUWP.vcxproj (hunk 1 above)
Comment (mganandraj) [reply vnext/ReactUWP/ReactUWP.vcxproj]: Ah .. thank you for pointing this out.. This was a test hermes nuget that i was using to test earlier.. I'll remove this entry.

**Comment by mganandraj** [comment]
> Do we want to enable it for ReactWin32 as well?



> Do we want to enable it for ReactWin32 as well?
All changes for win32 should be in our RNH layer in devmain.

**Pull request merged**